guides/plugins/apps/starter/add-api-endpoint.md

```diff
@@ -0,0 +1,320 @@
+{% endhint %}
+
+Next, we're gonna put our basic configuration into the file we just created.
+
+{% code title="manifest.xml" %}
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<manifest xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://raw.githubusercontent.com/shopware/platform/trunk/src/Core/Framework/App/Manifest/Schema/manifest-1.0.xsd">
+    <meta>
+        <name>MyApiExtension</name>
+        <label>Topsellers API</label>
+        <description>This app adds a Topseller API endpoint</description>
+        <author>shopware AG</author>
+        <copyright>(c) shopware AG</copyright>
+        <version>1.0.0</version>
+        <license>MIT</license>
+    </meta>
+    <permissions>
+        <read>order</read>
+        <read>order_line_item</read>
+        <read>product</read>
+    </permissions>
+</manifest>
+
+```
+{% endcode %}
+
+Besides some metadata, like a name, description or a version, this file contains permissions that the app requires.
+We will need them later on when performing searches.
+
+## Create the script
+
+We are going to define our new API endpoint in a script file based on [App Scripts](./../app-scripts/).
+There are specific directory conventions that we have to follow in order to register a new API endpoint script.
+The prefix for our API endpoint is one of the following and cannot be changed:
+
+| API        | API consumers / callers      | Prefix                |
+|------------|------------------------------|-----------------------|
+| Store API  | Customer-facing integrations | `/store-api/script/`  |
+| Admin API  | Backend integrations         | `/api/script/`        |
+| Storefront | Default Storefront           | `/storefront/script/` |
+
+{% hint style="info" %}
+You might wonder why the storefront shows up in that table. In storefront endpoints you can not only render JSON, but also twig templates.
+But use them with care - whenever you create a Storefront endpoint, your app will not be compatible with headless consumers.
+
+[Custom endpoints](../app-scripts/custom-endpoints.md)
+{% endhint %}
+
+### Directory structure
+
+In this example, we're going to create a Store API endpoint. We want to provide logic that returns the top-selling products for a specific category.
+So let's use the following endpoint naming:
+
+`/store-api/script/swag/topseller`
+
+You see that we've added a custom subdirectory `swag` in the route.
+This is a good practice, because we can prevent naming collisions between different apps. 
+Slashes (or subdirectories) in the endpoint path are represented by a hyphen in the name of the directory that contains the script.
+
+```
+MyApiExtension/
+├─ Resources/
+│  ├─ scripts/
+│  │  ├─ store-api-swag-topseller/ <-- /store-api/script/swag/topseller
```

Comment:
Just FYI `/store-api/script/swag-topseller` would also work as the URL with that filename, all `/` are replaced with `-` so both URLs would map